### PR TITLE
ICFY: Increase memory, allow failure

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

Let ICFY job report failure.
Increase ICFY memory size.

#### Testing instructions

Watch icfy job on CircleCI

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -461,13 +461,13 @@ jobs:
             # This block should not cause a test failure and block PRs.
             # The shell should never error and exit 0 to indicate success.
```
Comment (sgomes) [inline .circleci/config.yml]: Does this comment need updating?

Comment (sirreal) [inline .circleci/config.yml]:
```suggestion
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -459,15 +459,13 @@ jobs:
...
-            # The shell should never error and exit 0 to indicate success.
             #
             set +o errexit
-            npm run build-css                                      \
-              && npm run preanalyze-bundles                        \
-              && node bin/icfy-analyze.js                          \
-              && mv client/stats.json client/chart.json client/style.json "$CIRCLE_ARTIFACTS/icfy" \
+            npm run build-css                                                                       \
+              && npm run preanalyze-bundles                                                         \
+              && node --max-old-space-size=4096 bin/icfy-analyze.js                                 \
```
Comment (sirreal) [inline .circleci/config.yml]: Seems like a good size, these are the machine limits:

Screen Shot 2020-02-12 at 12 04 05

Might be in danger of passing machine limits, but we can let them fail if that happens.

**Comment by sirreal** [comment]
I see the failure build logged on ICFY from 9f69195.

**Comment by mmtr** [comment]
Seems the build is still failing silently:

```
242.2s 98% after emitting SizeLimitsPlugin
Webpack Bundle Analyzer saved stats file to /home/circleci/wp-calypso/client/stats.json
330.9s 100%  
Analyze: reading stats.json file
Analyze: analyzing
Analyze: writing chart.json file
Analyze: analyzing the style.css file
Analyze: finished
mv: cannot stat 'client/style.json': No such file or directory
```

**Comment by jsnajdr** [comment]
We haven't been producing the `style.json` file for many months, since the CSS migration to webpack was finished. It seems that the `mv` command copied `stats.json` and `chart.json` and then failed, and that the failure was ignored.

Let's remove the dead `style.json` code in #39409.

File: .circleci/config.yml [hunk 3]
```diff
@@ -459,15 +459,13 @@ jobs:
...
             #
             # This block should not cause a test failure and block PRs.
-            # The shell should never error and exit 0 to indicate success.
             #
             set +o errexit
-            npm run build-css                                      \
-              && npm run preanalyze-bundles                        \
-              && node bin/icfy-analyze.js                          \
-              && mv client/stats.json client/chart.json client/style.json "$CIRCLE_ARTIFACTS/icfy" \
+            npm run build-css                                                                       \
```
Comment (jsnajdr) [inline .circleci/config.yml]: We no longer need to run the `build-css` job, as it doesn't produce any stylesheed we'd want to analyze. We used to analyze `public/style.css`, but it was all migrated away to webpack.

**Comment by jsnajdr** [comment]
Rebased onto latest `master`, resolved conflicts with #39409, removed the `build-css` step. Let's see what CI says, but I think this is ready to 🚢 now.

**Comment by mmtr** [comment]
Is CircleCI still ignoring failures?

https://circleci.com/gh/Automattic/wp-calypso/599706

```
191.5s 92% chunk asset optimization 
191.5s 92% chunk asset optimization TerserPlugin
internal/child_process.js:394
    throw errnoException(err, 'spawn');
    ^

Error: spawn ENOMEM
    at ChildProcess.spawn (internal/child_process.js:394:11)
    at spawn (child_process.js:540:9)
    at Object.fork (child_process.js:108:10)
    at ChildProcessWorker.initialize (/home/circleci/wp-calypso/node_modules/jest-worker/build/workers/ChildProcessWorker.js:137:44)
    at ChildProcessWorker.onExit (/home/circleci/wp-calypso/node_modules/jest-worker/build/workers/ChildProcessWorker.js:263:12)
    at ChildProcess.emit (events.js:223:5)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12) {
  errno: 'ENOMEM',
  code: 'ENOMEM',
  syscall: 'spawn'
}
```

**Comment by jsnajdr** [comment]
> Is CircleCI still ignoring failures?

That's weird. ICFY didn't even receive a failure notification for this build. I.e., the `Notify ICFY of failed build` step didn't run.

**Comment by sirreal** [comment]
> > Is CircleCI still ignoring failures?
> 
> That's weird. ICFY didn't even receive a failure notification for this build. I.e., the `Notify ICFY of failed build` step didn't run.

It's the way the command is structured. It's chaining shell with `&&` and `||`, so the failure never surfaces to CircleCI.

File: .circleci/config.yml [hunk 4]
```diff
@@ -478,15 +478,12 @@ jobs:
           command: |
             #
             # This block should not cause a test failure and block PRs.
```
Comment (sirreal) [inline .circleci/config.yml]:
```suggestion
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -478,15 +478,12 @@ jobs:
...
-            # The shell should never error and exit 0 to indicate success.
             #
             set +o errexit
-            npm run build-css                                      \
-              && npm run preanalyze-bundles                        \
-              && node bin/icfy-analyze.js                          \
+            npm run preanalyze-bundles                                           \
+              && node --max-old-space-size=4096 bin/icfy-analyze.js              \
```
Comment (sirreal) [inline .circleci/config.yml]:
```suggestion
              node --max-old-space-size=4096 bin/icfy-analyze.js
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -478,15 +478,11 @@ jobs:
...
             set +o errexit
-            npm run build-css                                      \
-              && npm run preanalyze-bundles                        \
-              && node bin/icfy-analyze.js                          \
-              && mv client/stats.json client/chart.json "$CIRCLE_ARTIFACTS/icfy" \
-              || rm -fr build/.babel-client-cache # In case of failure do not save a potentially bad cache
-            exit 0
+            npm run preanalyze-bundles                                           \
+              && node --max-old-space-size=4096 bin/icfy-analyze.js              \
+              && mv client/stats.json client/chart.json "$CIRCLE_ARTIFACTS/icfy"
```
Comment (sirreal) [inline .circleci/config.yml]:
```suggestion
              mv client/stats.json client/chart.json "$CIRCLE_ARTIFACTS/icfy"
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -478,15 +478,11 @@ jobs:
...
-            # The shell should never error and exit 0 to indicate success.
             #
             set +o errexit
-            npm run build-css                                      \
-              && npm run preanalyze-bundles                        \
-              && node bin/icfy-analyze.js                          \
-              && mv client/stats.json client/chart.json "$CIRCLE_ARTIFACTS/icfy" \
-              || rm -fr build/.babel-client-cache # In case of failure do not save a potentially bad cache
-            exit 0
+            npm run preanalyze-bundles                                           \
```
Comment (sirreal) [inline .circleci/config.yml]:
```suggestion
            npm run preanalyze-bundles
```

File: .circleci/config.yml [hunk 8]
```diff
@@ -478,15 +478,11 @@ jobs:
           command: |
             #
```
Comment (sirreal) [inline .circleci/config.yml]:
```suggestion
```

File: .circleci/config.yml [hunk 9]
```diff
@@ -478,15 +478,11 @@ jobs:
...
             # This block should not cause a test failure and block PRs.
-            # The shell should never error and exit 0 to indicate success.
             #
```
Comment (sirreal) [inline .circleci/config.yml]:
```suggestion
```

**Review by sirreal — COMMENTED** [review]
Minor: I don't think we need to chain these commands now. Any individual failure should fail and end the step.

File: .circleci/config.yml [hunk 10]
```diff
@@ -476,17 +476,11 @@ jobs:
...
-            npm run build-css                                      \
-              && npm run preanalyze-bundles                        \
-              && node bin/icfy-analyze.js                          \
-              && mv client/stats.json client/chart.json "$CIRCLE_ARTIFACTS/icfy" \
-              || rm -fr build/.babel-client-cache # In case of failure do not save a potentially bad cache
-            exit 0
+            npm run preanalyze-bundles
+            node --max-old-space-size=4096 bin/icfy-analyze.js
+            mv client/stats.json client/chart.json "$CIRCLE_ARTIFACTS/icfy"
+            exit 1
```
Comment (sirreal) [inline .circleci/config.yml]:
```suggestion
```

**Comment by sirreal** [comment]
In https://circleci.com/gh/Automattic/wp-calypso/599962 we see that when we let the step and job fail, no caches are saved. We wanted to confirm that 👍

**Comment by sirreal** [comment]
I'd say this is working well ✅

**Review by mmtr — APPROVED** [review]
Thanks for jumping into this!

**Review by jsnajdr — APPROVED** [review]
Alright @sirreal, do you want to merge this? Thanks for a productive collaboration 🤝

**Pull request merged**